CMS-Enterprise/easi-app
# [EASI-4486] system intake doc uploads

# EASI-4486

## Description


- guard document actions (create, view, delete) to allow only those authorized to make changes

## How to test this change

- sign in with a test EUA with only the `EASI_P_USER` role
- create a system intake and attach a document (see note)
- confirm the document creates (this confirms a REQUESTER can create a doc)
- view the document (this confirms a REQUESTER can view a doc)
- log out
- log in with a different EUA and select `EASI_D_GOVTEAM` role
- navigate to the original system intake
- confirm you can view the document (this confirms an ADMIN can view a doc)
- attempt to delete the doc - it should fail (this confirms an ADMIN cannot delete a REQUESTER doc)
- log out
- log back in with original user with `EASI_P_USER` role
- go try to delete the doc - it should succeed (this confirms a REQUESTER can delete their own doc)




> [!NOTE]
> i forced the virus scan return by doing this
```go
// GetStatusForSystemIntakeDocument queries S3 for the virus-scanning status of a document with the given s3Key
func GetStatusForSystemIntakeDocument(s3Client *upload.S3Client, s3Key string) (models.SystemIntakeDocumentStatus, error) {
	avStatus, err := s3Client.TagValueForKey(s3Key, upload.AVStatusTagName)
	if err != nil {
		return "", err
	}

	return models.SystemIntakeDocumentStatusAvailable, nil

	// possible tag values come from virus scanning lambda
	if avStatus == "CLEAN" {
		return models.SystemIntakeDocumentStatusAvailable, nil
	} else if avStatus == "INFECTED" {
		return models.SystemIntakeDocumentStatusUnavailable, nil
	} else {
		return models.SystemIntakeDocumentStatusPending, nil
	}
}
```


## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
